src/spartan_tcp_handler.erl

```diff
@@ -39,7 +39,6 @@ callback_mode() ->
     state_functions.
 
 init([Ref, Socket, Transport, Opts]) ->
-    {ok, _} = timer:kill_after(?TIMEOUT * 5),
```

Comment:
I found only one place without timeout and added it there: https://github.com/dcos/spartan/pull/42/files#diff-d255b854425e892855c602c97dc593d0L106

Removing of timeouts is the most complicated part of this PR. I would be so grateful if someone tries to find problems with it.